streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -1249,11 +1248,7 @@ public void cleanUp() {
 
         log.debug("Current changelog positions: {}", allChangelogPositions);
         final Map<TopicPartition, ListOffsetsResultInfo> allEndOffsets;
-        try {
-            allEndOffsets = fetchEndOffsets(allPartitions, adminClient);
-        } catch (final TimeoutException e) {
-            throw new StreamsException("Timed out obtaining end offsets from kafka", e);
-        }
+        allEndOffsets = fetchEndOffsets(allPartitions, adminClient);
```

Comment:
Minor side fix: `fetchEndOffset` can never throw `TimeoutException` because it catches all `RuntimeException` and convert them into `StreamsException` already